### ISSUE TITLE
Refactor to mobx-state-tree

### ISSUE THREAD

**Issue by Entryist** [getZION/app#9]
Finish the `mobx-state-tree` refactor we've been working on since September.

In our initial attempt to complete the MST refactor for 1.15, we ran into issues not addressed by the switch from vanilla mobx to mobx-state-tree:

* Messy message handling (with excessive iteration) causing lag for users with lots of messages - fixed in 1.15
* Issues with media messages not showing for communities - fixed in 1.16

Now that core functionality is working more-or-less smoothly for users with large amounts of messages, we are ready to finalize the MST refactor, which will put us on a much stronger foundation for future features.

Our pre-1.15 refactor work swapped out the old vanilla stores for new MST stores. After connecting the original app components to the new stores, we realized that a large source of the lag stems from excessive iteration in the components' hooks that filter and sort messages.

To summarize the work remaining to be done, the hooks need to be refactored to reflect more sophisticated algorithms for fetching/decrypting/storing messages, and the 100+ components need to be updated to use those new hooks.

**Comment by Entryist** [comment]
Solved in 1.17
